packages/grafana-ui/src/components/PanelChrome/PanelChrome.tsx

```diff
@@ -9,12 +9,21 @@ import { useStyles2, useTheme2 } from '../../themes';
  * @internal
  */
 export interface PanelChromeProps {
+  title?: string;
+  items?: {
+    orderedList: ReactNode[];
+    position?: 'left' | 'right'; // Default is 'left'
+  };
+  actionItems?: {
+    orderedList: ReactNode[];
+    position?: 'left' | 'right'; // Default is 'left'
+    show?: 'always' | 'hover' | 'never'; // Default is 'hover'
+  };
   width: number;
   height: number;
-  title?: string;
   padding?: PanelPadding;
-  leftItems?: React.ReactNode[]; // rightItems will be added later (actions links etc.)
-  children: (innerWidth: number, innerHeight: number) => React.ReactNode;
+  leftItems?: ReactNode[]; // rightItems will be added later (actions links etc.)
+  children: (innerWidth: number, innerHeight: number) => ReactNode;
```

Comment:
think this feels a bit too complex (and nested), The show "hover" , and position props on items, and actions items feel wrong (esp given we already have a leftItems) 

think something like this 
```ts
export interface PanelChromeProps {
  width: number;
  height: number;
  title?: string;
  padding?: PanelPadding;
  dragClass?: string;  // set if draggable, new grid might need different drag handle classes for each nested grid
  menu?: Menu;   // Panel menu actions 
  hoverHeader?: boolean; // Show header only on hover or when panel has focus (absolutely positioned) 
  leftItems?: React.ReactNode[]; // Info icons/states to left of title, if hoverHeader shown in the hover header
  loadingState?: LoadingState; // If we want to show the loading state via animated border 
  states? React.ReactNode[] // always visible loading and error states in the right corner, always shown in panel right corner, even in hover mode)
  children: (innerWidth: number, innerHeight: number) => React.ReactNode;
}
```

Reply:
I am not 100% sure what the latest design is, but I think we should come up with concepts that are a bit agnostic on design so we can change the design without changing how consumers use PanelChrome

Reply:
Okay, this makes sense - it is a bit too complex the way I tested it out. I like the `hoverHeader`, `menu`, `states`, `loadingState`, and `dragClass` props. 

However, `leftItems` already exists with the purpose of holding the states on the right, we have to go through the process of deprecating it? 
Also, it doesn't seem necessary to me that we have `leftItems` to be placed on the left of the title when there's only one thing to be placed there - the drag handle if `hoverHeader === true`. Maybe we should we just place it there?

We will need some props for the icons next to the title, some of which are always visible and some (the menu) are visible on hover. I will try out more straightforward approach with no nesting and less positioning "flexibility".

Reply:
We could deprecate leftItems and add a new more typed concept, an array of i something like: 

```ts
interface PanelChromeInfoState  {
   icon: string;
   tooltip: string;
   onClick: () => {}
}
```

> Also, it doesn't seem necessary to me that we have leftItems to be placed on the left of the title when there's only one thing to be placed there -

not sure what you mean? There is nothing to the left of the title. Only to the right. (My comment was wrong, meant right of title).

Reply:
`PanelChromeInfoState` defines the necessary information for the render of the actionable icon, but an array of those would still be nested. It is as good as it gets for now, however, thank you for the tips! 

> There is nothing to the left of the title. Only to the right.

I meant the drag-n-drop "6 dots" icon handle that shows up on the left of the title when the title is hidden as a whole.